packages/react-native-codegen/src/parsers/flow/modules/index.js

```diff
@@ -68,6 +68,26 @@ function nullGuard<T>(fn: () => T): ?T {
   return fn();
 }
 
+const UnsupportedObjectPropertyTypeToInvalidPropertyValueTypeMap = {
+  FunctionTypeAnnotation: 'FunctionTypeAnnotation',
+  VoidTypeAnnotation: 'void',
+  PromiseTypeAnnotation: 'Promise',
+};
+
+function throwUnsupportedObjectPropertyValueTypeAnnotationParserError(
+  moduleName,
+  propertyValue,
+  propertyKey,
+  type,
+) {
+  throw new UnsupportedObjectPropertyValueTypeAnnotationParserError(
```

Comment:
I will put also the `if` logic in the function, s we can factor out more code.

Also, we can have a better name. Something like: `throwIfPropertyValueTypeIsUnsupported`. What do you think?